### DISCUSSION TITLE
Is this Unity Asset official?

### DISCUSSION THREAD

**Discussion by alloystorm** [rhasspy/piper#201] [General]
This Unity Asset is based on piper
https://assetstore.unity.com/packages/tools/audio/speech-generation-system-offline-text-to-speech-conversion-255039

I bought it and it works fine. The only thing is that there's no way to set speaker_id, so only the first speaker is used. 
I'm wondering if this asset is official? If so is there future plan to add support for speaker id?

**Comment by synesthesiam** [comment]
Not an official asset since I only just heard of it 😄
Hopefully they add the ability to set speaker id.

**Comment by alloystorm** [comment]
Thanks for you reply.
 
Sorry if this is a dumb question but I'm just wondering how much of the heavy lifting is in the cpp code and how much is in the model itself?

Seeing that Unity has a new Sentis thing coming out that they claim supports inference of onnx models on multiple platforms including mobile, do you think it's possible to get the piper tts models run on that as well?

**Reply by synesthesiam** [reply]
Not a dumb question at all. Unfortunately, a lot of heavy lifting is done by the C++ code (really by espeak-ng) for most voice models. This is because the voices are trained on phonemes (fundamental units of human speech) rather than raw text.

It's possible to train models on raw text with Piper, but you then need a preprocessing stage to spell out numbers, dates, etc. (e.g., 1 to "one"). It's not easy in general, since text like 1/4 could be a fraction or date depending on context.
